### PR TITLE
Don't upload artifacts to s3 on pull_quests

### PR DESCRIPTION
**Pull request by jooola**
I am not totally sure about this. Do liquidsoap rely on pull_requests for uploading artifacts ?

### CODE REVIEW

**Comment by toots** [comment]
Thanks. As I said in the other ticket, s3 upload is a work around the limitations of the standard upload action on arm64 containers.

**Comment by jooola** [comment]
Yes, but I was wondering if there is a usecase for uploading the artifact that were build in pull_request. I would expect to build and upload from the main/v*/rolling branches, but not every pull request.

Long story short, should we disable uploading artifact for pull_requests ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -238,14 +238,14 @@ jobs:
           cd /tmp/liquidsoap-full/liquidsoap
           sudo -u opam -E ./.github/scripts/build-deb.sh ${{ github.sha }} ${{ needs.build_details.outputs.branch }} ${{ matrix.os }} ${{ matrix.platform }} "${{ needs.build_details.outputs.is_rolling_release }}" "${{ needs.build_details.outputs.is_release }}"
       - name: Upload debian packages artifacts
-        if: (contains(matrix.os, 'debian') || contains(matrix.os, 'ubuntu')) && matrix.platform != 'armhf'
+        if: github.repository_owner == 'savonet' && github.event_name == 'push' && (contains(matrix.os, 'debian') || contains(matrix.os, 'ubuntu')) && matrix.platform != 'armhf'
```
Comment (toots) [inline .github/workflows/ci.yml:241]:
```suggestion
        if: needs.build_details.outputs.is_release && (contains(matrix.os, 'debian') || contains(matrix.os, 'ubuntu')) && matrix.platform != 'armhf'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -238,14 +238,14 @@ jobs:
...
         uses: actions/upload-artifact@v2
         with:
           name: ${{ steps.build_deb.outputs.basename }}
           path: ${{ github.workspace }}/${{ github.run_number }}/${{ matrix.os }}_${{ matrix.platform }}/debian
           if-no-files-found: error
       - name: Upload debian armhf packages artifacts
-        if: github.repository_owner == 'savonet' && contains(matrix.os, 'debian') && matrix.platform == 'armhf'
+        if: github.repository_owner == 'savonet' && github.event_name == 'push' && contains(matrix.os, 'debian') && matrix.platform == 'armhf'
```
Comment (toots) [inline .github/workflows/ci.yml:248]:
```suggestion
        if: needs.build_details.outputs.is_release && contains(matrix.os, 'debian') && matrix.platform == 'armhf'
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -263,7 +263,7 @@ jobs:
           chown -R opam /tmp/${{ github.run_number }}/${{ matrix.os }}_${{ matrix.platform }}/alpine
           sudo -u opam -E ./.github/scripts/build-apk.sh ${{ needs.build_details.outputs.branch }} ${{ matrix.os }} ${{ matrix.platform }} ${{ matrix.alpine-arch }} "${{ needs.build_details.outputs.is_rolling_release }}" "${{ needs.build_details.outputs.is_release }}" ${{ github.sha }}
       - name: Upload alpine packages artifacts
-        if: github.repository_owner == 'savonet' && matrix.os == 'alpine'
+        if: github.repository_owner == 'savonet' && github.event_name == 'push' && matrix.os == 'alpine'
```
Comment (toots) [inline .github/workflows/ci.yml:266]:
```suggestion
        if: needs.build_details.outputs.is_release && matrix.os == 'alpine'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -279,7 +279,7 @@ jobs:
     runs-on: ubuntu-latest
     needs: [build_details, build_posix]
 
-    if: github.repository_owner == 'savonet'
+    if: github.repository_owner == 'savonet' && github.event_name == 'push'
```
Comment (toots) [inline .github/workflows/ci.yml:282]:
```suggestion
    if: needs.build_details.outputs.is_release
```

**Comment by toots** [comment]
Ok, I factored out the logic to have an explicit `is_fork` boolean and make sure a fork branch is never tagged for release. That clarifies the internal logic.

**Comment by jooola** [comment]
I use this to not go mad while debugging github actions https://github.com/mxschmitt/action-tmate

**Comment by toots** [comment]
Well turns out there doesn't seem to be an easy way to detect is a branch is from a fork.. Repository owner is `savonet` all the time. I decided to use the availability of the S3 secrets. Good enough for now.

**Comment by jooola** [comment]
https://github.com/orgs/community/discussions/25217#discussioncomment-3246902

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -40,7 +40,7 @@ jobs:
         uses: actions/checkout@v3
       - name: Get build details
         env:
-          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          IS_FORK: ${{ github.event_name == 'pull_request' && (github.event.pull_request.head.repo.full_name == github.repository) }}
```
Comment (jooola) [inline .github/workflows/ci.yml:43]: I found a cleaner way: github.event.pull_request.head.repo.fork

**Pull request merged**